Aula 7 (#10)
lucianopf/git-workshop: lessons/lesson-7.md

# Aula 7 (fetch, pull)

> Quando trabalhamos em conjunto com outras pessoas é comum o estado remoto do repositório mudar enquanto estamos trabalhando, então precisamos atualizar o nosso repositório local com as mudanças do repositório remoto.

Reviewer: ```suggestion
Quando trabalhamos em conjunto com outras pessoas, é comum o estado remoto do repositório mude. Então, precisamos atualizar o nosso repositório local com as mudanças do repositório remoto.
```

Pra isso podemos usar os comandos `fetch` e `pull`.

**No fim desse arquivo teremos um exercício prático antes de seguir para a próxima aula.**

## Fetch

> O comando `fetch` nos permite atualizar o estado do repositório local com o estado do repositório remoto sem fazer merge com a branch atual.

Reviewer: ```suggestion
O comando `fetch` nos permite atualizar o estado local com o estado remoto, sem realizar `merge` ou `rebase` na branch atual.
```

> Assim como os demais comandos o `git fetch` também tem suas variações e a que mais uso é a `git fetch --all` que garante que atualizamos todas as referencias de todas os repositórios remotos que temos.

Reviewer: ```suggestion
Assim como os demais comandos, o `git fetch` também tem suas variações. A que mais utilizo, é a `git fetch --all`, que garante a atualização de todas as referencias entre nosso repositório local e o remoto.
```

> Obs: apesar de não termos abordado até o momento é possível e bem comum quando colaborando com open-source ter mais de um repositório remoto, por exemplo, o repositório original e o seu fork.

Reviewer: ```suggestion
Obs: apesar de não termos abordado até o momento, é possível e bem comum termos mais de um repositório remoto. Por exemplo, o original e o seu `for`. 
```
